tools/loader/ImageClassifier.cpp

```diff
@@ -153,68 +120,46 @@ llvm::cl::opt<bool> useImagenetNormalization(
     llvm::cl::cat(imageLoaderCat), llvm::cl::init(false));
 } // namespace
 
+/// Insert \p image into \p batch at \p index.
+template <typename Ty>
+void insertImage(Handle<Ty> &batch, const Handle<Ty> &image, size_t index) {
+  auto dims = image.dims();
+  for (size_t i = 0; i < dims[0]; i++) {
+    for (size_t j = 0; j < dims[1]; j++) {
+      for (size_t k = 0; k < dims[2]; k++) {
+        batch.at({index, i, j, k}) = image.at({i, j, k});
+      }
+    }
+  }
+}
+
 /// Loads and normalizes all PNGs into a tensor in the NHWC format with the
 /// requested channel ordering.
 void loadImagesAndPreprocess(const llvm::cl::list<std::string> &filenames,
                              Tensor *inputImageData) {
   assert(!filenames.empty() &&
          "There must be at least one filename in filenames.");
-  auto range = normModeToRange(imageNormMode);
   unsigned numImages = filenames.size();
 
-  // Get first image's dimensions and check if grayscale or color.
-  size_t imgHeight, imgWidth;
-  bool isGray;
-  std::tie(imgHeight, imgWidth, isGray) = getPngInfo(filenames[0].c_str());
-  const size_t numChannels = isGray ? 1 : 3;
-
-  // N x C x H x W
-  inputImageData->reset(ElemKind::FloatTy,
-                        {numImages, numChannels, imgHeight, imgWidth});
+  Tensor initial =
```

Comment:
Curious, why did you decide to move loading the first image separately to set the dims of `inputImageData`? Before we just called `getPngInfo()` on the first image in order to set the dimensions of `inputImageData`, and then loaded all images in a loop. I guess it's one less `fopen()` this way? If you prefer this way then I think we could actually get rid of `getPngInfo()` entirely...